contracts/DebtToken.sol

```diff
@@ -155,93 +173,86 @@ contract DebtToken is Ownable {
   /**
   Check if updateInterest() needs to be called before refundLoan()
   */
-  function interestStatusUpdated() public constant returns(bool){
-    if(!loanMature())
+  function isInterestStatusUpdated() public constant returns(bool){
+    if(!isTermOver())
       return true;
     else
-      return !( now >= lastinterestCycle.add( interestCycleLength.mul(dayLength) ) );
+      return !( now >= lastInterestCycle.add( interestCycleLength.mul(dayLength) ) );
   }
 
-  /**
-
-  */
-
   /**
   calculate the total number of passed interest cycles and coin value
   */
   function calculateInterestDue() public constant returns(uint256 _coins,uint256 _cycle){
-    if(!loanMature())
-      return (0,0);
-    else if(balances[debtOwner] == 0)
+    if(!isTermOver() || !isLoanFunded())
```

Comment:
`!isLoanFunded()` use on this line is redundant, as `!isTermOver()` has implicit check for isFunded

Reply:
Agree